### PR TITLE
Combine condition check in track

### PR DESCRIPTION
**Pull request by darionyaphet**

### CODE REVIEW

**Review by jay-zhuang — APPROVED** [review]

File: trace_replay/trace_replay.cc [hunk 1]
```diff
@@ -92,10 +92,8 @@ Status TracerHelper::DecodeTrace(const std::string& encoded_trace,
                                  Trace* trace) {
   assert(trace != nullptr);
   Slice enc_slice = Slice(encoded_trace);
-  if (!GetFixed64(&enc_slice, &trace->ts)) {
-    return Status::Incomplete("Decode trace string failed");
-  }
-  if (enc_slice.size() < kTraceTypeSize + kTracePayloadLengthSize) {
+  if (!GetFixed64(&enc_slice, &trace->ts) ||
+      enc_slice.size() < kTraceTypeSize + kTracePayloadLengthSize) {
```
Comment (riversand963) [inline trace_replay/trace_replay.cc:96]: Should we instead report different error messages here?

**Review by riversand963 — CHANGES_REQUESTED** [review]
See comments.

File: trace_replay/trace_replay.cc (hunk 1 above)
Comment (darionyaphet) [reply trace_replay/trace_replay.cc:96]: Actually, They're all failure about decode. I think it's fine . WDYT ~

Comment (riversand963) [reply trace_replay/trace_replay.cc:96]: I prefer providing more information about the cause of failure when it's feasible and inexpensive.

**Comment by jay-zhuang** [comment]
close as no response.

**Pull request closed without merging**